internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -96,6 +124,7 @@ func (t *Translator) ProcessBackendTrafficPolicies(resources *resource.Resources
 				// policy overrides and populate its ancestor status.
 				parentRefs := GetParentReferences(route)
 				ancestorRefs := make([]gwapiv1a2.ParentReference, 0, len(parentRefs))
+				routeGateways := sets.New[types.NamespacedName]()
```

Comment:
nit: can we call this `routeParents` to make it easier to understand ?

Reply:
can't we reuse parentRefs ? (trying to think of ways to reduce men)

Reply:
if all the parents are gateway, we can reuse it.

Reply:
got it, let's keep the extra var